Mathlib/Combinatorics/SimpleGraph/Connectivity.lean

```diff
@@ -2612,6 +2612,19 @@ instance : Decidable G.Connected := by
   rw [connected_iff, ← Finset.univ_nonempty_iff]
   infer_instance
 
+open Finset in
+instance Path.instFintype {u v : V} : Fintype (G.Path u v) where
+  elems := (range (Fintype.card V)).biUnion
+    fun n ↦ (univ (α := {p : G.Walk u v | p.IsPath ∧ p.length = n})).image
+      fun p ↦ { val := p.val, property := p.prop.left }
```

Comment:
This is going to be inefficient for computation, I think; something like
```suggestion
  elems := (range (Fintype.card V)).disjiUnion
    (fun n ↦ (univ (α := {p : G.Walk u v | p.IsPath ∧ p.length = n})).map
      ⟨fun p ↦ { val := p.val, property := p.prop.left }, sorry⟩) sorry
```
would behave much better.

Of course, this has the downside that you have two proofs to fill!

Reply:
Sounds good. Out of curiosity, where would this inefficiency show? Is it just that we are avoiding the two extra deduplication steps during construction of the `Finset`?

Reply:
It would come when evaluating some computable sum over all paths, which would then perform all this deduplication at runtime. It would also appear if you evaluate something like `example : Fintype.card (G.Path _ _) = 10 = rfl` for some computable `G`.